### ISSUE TITLE
Bump Wazuh dashboard to OpenSearch Dashboards 2.10.0

### ISSUE THREAD

**Issue by rauldpm** [wazuh/wazuh-packages#2521]
## Description

It is necessary to adapt the Wazuh dashboard to version 2.10.0 of OpenSearch Dashboards
Request: https://github.com/wazuh/internal-devel-requests/issues/301

## Tasks

- [x] Bump OpenSearch Dashboards version https://github.com/wazuh/wazuh-packages/blob/master/stack/dashboard/base/builder.sh#L19
- [x] Build the package locally
  - [x] Base 
  - [x] RPM
  - [x] DEB
 - [x] Test the packages locally
   - [x] RPM
   - [x] DEB
 - [x] Modify https://github.com/wazuh/wazuh-packages/blob/master/README.md matrix with the new version 
 - [x] Create ECS and ECR resources (base, rpm, deb)
 - [x] Test the package using the Test_stack pipeline https://ci.wazuh.info/job/Test_stack_tier/97/
 - [x] Test the package using the Wazuh installation assistant

## Validation

- [x] The package presents normal operation and without errors

---

### Working branch

- [enhancement/2521-bump-dashboard](https://github.com/wazuh/wazuh-packages/tree/enhancement/2521-bump-dashboard)

**Comment by rauldpm** [comment]
## Update report - Build Base

- Bumped Readme to 2.10.0
- Bumped base to 2.10.0
- The base cannot be built because the APP cannot be generated
  - The [4.8.0 wazuh-dashboard-plugins branch](https://github.com/wazuh/wazuh-dashboard-plugins/blob/1bc2991c3b343abda750de4d04c15bf53816046f/plugins/main/scripts/release/tag.js#L64) references to OpenSearch 2.9.0
- Apparently, neither RPM nor DEB make reference to versioned files, due to the large number of files that the OpenSearch Dashboards base introduces, it is not possible to make a manual comparison, it is necessary that the construction of the package itself identifies which files are missing, which ones are present and which are duplicates, but for this it is necessary to build the APP
- Since the APP cannot be made, this issue goes to Blocked status, the @wazuh/frontend team has been asked about the problem and if it is possible to create a basic APP

**Comment by rauldpm** [comment]
## Analysis report - RPM

- Created DEB and RPM Wazuh manager with wp2521 revision
  - https://ci.wazuh.info/view/Packages/job/Packages_builder_tier/3189/
  - https://packages-dev.wazuh.com/warehouse/test/4.8/rpm/var/wazuh-manager-4.8.0-wp2521.x86_64.rpm
  - https://packages-dev.wazuh.com/warehouse/test/4.8/deb/var/wazuh-manager_4.8.0-wp2521_amd64.deb
- Installed Wazuh indexer and Wazuh server with wp2521 revision

---

<details><summary>Install :green_circle: </summary>

```
[root@centos7 vagrant]# yum localinstall https://packages-dev.wazuh.com/warehouse/test/4.8/rpm/var/wazuh-dashboard-4.8.0-wp2521.x86_64.rpm
Loaded plugins: fastestmirror
wazuh-dashboard-4.8.0-wp2521.x86_64.rpm                                                                                                                                                     | 269 MB  00:00:12     
Examining /var/tmp/yum-root-EwxXS9/wazuh-dashboard-4.8.0-wp2521.x86_64.rpm: wazuh-dashboard-4.8.0-wp2521.x86_64
Marking /var/tmp/yum-root-EwxXS9/wazuh-dashboard-4.8.0-wp2521.x86_64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package wazuh-dashboard.x86_64 0:4.8.0-wp2521 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

===================================================================================================================================================================================================================
 Package                                         Arch                                   Version                                         Repository                                                            Size
===================================================================================================================================================================================================================
Installing:
 wazuh-dashboard                                 x86_64                                 4.8.0-wp2521                                    /wazuh-dashboard-4.8.0-wp2521.x86_64                                 891 M

Transaction Summary
===================================================================================================================================================================================================================
Install  1 Package

Total size: 891 M
Installed size: 891 M
Is this ok [y/d/N]: y
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : wazuh-dashboard-4.8.0-wp2521.x86_64                                                                                                                                                             1/1 
  Verifying  : wazuh-dashboard-4.8.0-wp2521.x86_64                                                                                                                                                             1/1 

Installed:
  wazuh-dashboard.x86_64 0:4.8.0-wp2521                                                                                                                                                                            

Complete!
```

</details>  

<details><summary>Certificate copy :green_circle: </summary>

```
[root@centos7 vagrant]# NODE_NAME=dashboard-1
[root@centos7 vagrant]# mkdir /etc/wazuh-dashboard/certs
[root@centos7 vagrant]# tar -xf ./wazuh-certificates.tar -C /etc/wazuh-dashboard/certs/ ./$NODE_NAME.pem ./$NODE_NAME-key.pem ./root-ca.pem
[root@centos7 vagrant]# mv -n /etc/wazuh-dashboard/certs/$NODE_NAME.pem /etc/wazuh-dashboard/certs/dashboard.pem
[root@centos7 vagrant]# mv -n /etc/wazuh-dashboard/certs/$NODE_NAME-key.pem /etc/wazuh-dashboard/certs/dashboard-key.pem
[root@centos7 vagrant]# chmod 500 /etc/wazuh-dashboard/certs
[root@centos7 vagrant]# chmod 400 /etc/wazuh-dashboard/certs/*
[root@centos7 vagrant]# chown -R wazuh-dashboard:wazuh-dashboard /etc/wazuh-dashboard/certs
```

</details>

<details><summary>Start service :red_circle: </summary>

```
[root@centos7 vagrant]# systemctl daemon-reload
[root@centos7 vagrant]# systemctl enable wazuh-dashboard
Created symlink from /etc/systemd/system/multi-user.target.wants/wazuh-dashboard.service to /etc/systemd/system/wazuh-dashboard.service.
[root@centos7 vagrant]# systemctl start wazuh-dashboard
[root@centos7 vagrant]# systemctl status wazuh-dashboard.service 
● wazuh-dashboard.service - wazuh-dashboard
   Loaded: loaded (/etc/systemd/system/wazuh-dashboard.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Tue 2023-10-17 17:38:02 UTC; 5s ago
  Process: 6830 ExecStart=/usr/share/wazuh-dashboard/bin/opensearch-dashboards -c /etc/wazuh-dashboard/opensearch_dashboards.yml (code=exited, status=1/FAILURE)
 Main PID: 6830 (code=exited, status=1/FAILURE)

Oct 17 17:38:00 centos7 systemd[1]: Started wazuh-dashboard.
Oct 17 17:38:02 centos7 opensearch-dashboards[6830]: {"type":"log","@timestamp":"2023-10-17T17:38:02Z","tags":["info","plugins-service"],"pid":6830,"message":"Plugin \"dataSourceManagement\" has ...dataSource]"}
Oct 17 17:38:02 centos7 opensearch-dashboards[6830]: {"type":"log","@timestamp":"2023-10-17T17:38:02Z","tags":["info","plugins-service"],"pid":6830,"message":"Plugin \"dataSource\" is disabled."}
Oct 17 17:38:02 centos7 opensearch-dashboards[6830]: {"type":"log","@timestamp":"2023-10-17T17:38:02Z","tags":["info","plugins-service"],"pid":6830,"message":"Plugin \"visTypeXy\" is disabled."}
Oct 17 17:38:02 centos7 opensearch-dashboards[6830]: {"type":"log","@timestamp":"2023-10-17T17:38:02Z","tags":["fatal","root"],"pid":6830,"message":"Error: listen EACCES: permission denied 0.0.0.0:443\n    at...
Oct 17 17:38:02 centos7 opensearch-dashboards[6830]: FATAL  Error: listen EACCES: permission denied 0.0.0.0:443
Oct 17 17:38:02 centos7 systemd[1]: wazuh-dashboard.service: main process exited, code=exited, status=1/FAILURE
Oct 17 17:38:02 centos7 systemd[1]: Unit wazuh-dashboard.service entered failed state.
Oct 17 17:38:02 centos7 systemd[1]: wazuh-dashboard.service failed.
Hint: Some lines were ellipsized, use -l to show in full.
```

</details>

**Comment by rauldpm** [comment]
On hold due https://github.com/wazuh/wazuh/issues/19779

**Comment by rauldpm** [comment]
## Update report - Build RPM

- Improved SPEC files in order to reduce the %files list using *
- Package build successfully, warning about duplicate node binaries (listed twice) no longer appears
- Package installed successfully
- The changes of the has not been merged yet https://github.com/wazuh/wazuh-packages/pull/2540, these changes can introduce new changes in the SPECS files as the base file has been modified

**Comment by rauldpm** [comment]
## Update report

- After talking with the @wazuh/frontend team, it was clear that the wazuh-dashboard-app repository does not have the necessary changes according to the merged changes in PR https://github.com/wazuh/wazuh-packages/pull/2540
- The main issue is On hold https://github.com/wazuh/wazuh-dashboard/issues/94, but they provided a draft PR https://github.com/wazuh/wazuh-dashboard-plugins/pull/5840
  - A new APP is being built with this custom branch
  - Created 4.8.0 wazuhapp ECR image: dashboardapp_builder_noarch_4-8-0
  - Created 4.8.0 ECS Task Definition: Packages_dashboardapp_noarch_4-8-0:1
  - Created new APP: https://ci.wazuh.info/view/Packages/job/Packages_builder/171271

**Comment by rauldpm** [comment]
## Update report

- Fixed all logo changes
- Tested functionality with the latest changes
- Improved base builder and adapted OpenSearch internal changes
  -  https://github.com/wazuh/wazuh-packages/commit/34f6de750070296772693be2ac647864825bcb79
  - https://github.com/wazuh/wazuh-packages/commit/48d3d62daee0b54c475bff3249c9d9e75165cd47

**Comment by rauldpm** [comment]
## Analysis report - RPM upgrade

<details><summary>Install 4.5.4 Stack</summary>

```
[root@centos7 vagrant]# curl -sO https://packages.wazuh.com/4.5/wazuh-install.sh && sudo bash ./wazuh-install.sh -a
26/10/2023 12:47:31 INFO: Starting Wazuh installation assistant. Wazuh version: 4.5.4
26/10/2023 12:47:31 INFO: Verbose logging redirected to /var/log/wazuh-install.log
26/10/2023 12:47:35 INFO: --- Dependencies ---
26/10/2023 12:47:35 INFO: Installing lsof.
26/10/2023 12:47:41 INFO: Wazuh web interface port will be 443.
26/10/2023 12:47:42 INFO: Wazuh repository added.
26/10/2023 12:47:42 INFO: --- Configuration files ---
26/10/2023 12:47:42 INFO: Generating configuration files.
26/10/2023 12:47:42 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
26/10/2023 12:47:42 INFO: --- Wazuh indexer ---
26/10/2023 12:47:42 INFO: Starting Wazuh indexer installation.
26/10/2023 12:48:26 INFO: Wazuh indexer installation finished.
26/10/2023 12:48:26 INFO: Wazuh indexer post-install configuration finished.
26/10/2023 12:48:26 INFO: Starting service wazuh-indexer.
26/10/2023 12:48:33 INFO: wazuh-indexer service started.
26/10/2023 12:48:33 INFO: Initializing Wazuh indexer cluster security settings.
26/10/2023 12:48:43 INFO: Wazuh indexer cluster initialized.
26/10/2023 12:48:43 INFO: --- Wazuh server ---
26/10/2023 12:48:43 INFO: Starting the Wazuh manager installation.
26/10/2023 12:49:05 INFO: Wazuh manager installation finished.
26/10/2023 12:49:05 INFO: Starting service wazuh-manager.
26/10/2023 12:49:19 INFO: wazuh-manager service started.
26/10/2023 12:49:19 INFO: Starting Filebeat installation.
26/10/2023 12:49:22 INFO: Filebeat installation finished.
26/10/2023 12:49:22 INFO: Filebeat post-install configuration finished.
26/10/2023 12:49:22 INFO: Starting service filebeat.
26/10/2023 12:49:22 INFO: filebeat service started.
26/10/2023 12:49:22 INFO: --- Wazuh dashboard ---
26/10/2023 12:49:22 INFO: Starting Wazuh dashboard installation.
26/10/2023 12:50:05 INFO: Wazuh dashboard installation finished.
26/10/2023 12:50:05 INFO: Wazuh dashboard post-install configuration finished.
26/10/2023 12:50:05 INFO: Starting service wazuh-dashboard.
26/10/2023 12:50:05 INFO: wazuh-dashboard service started.
26/10/2023 12:50:19 INFO: Initializing Wazuh dashboard web application.
26/10/2023 12:50:19 INFO: Wazuh dashboard web application initialized.
26/10/2023 12:50:19 INFO: --- Summary ---
26/10/2023 12:50:19 INFO: You can access the web interface https://<wazuh-dashboard-ip>:443
    User: admin
    Password: KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA
26/10/2023 12:50:19 INFO: Installation finished.
```

</details>

<details><summary>Check 4.5.4 Stack</summary>

```
[root@centos7 vagrant]# curl -k -u admin:KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA https://localhost:9200/_cat/nodes?v
ip        heap.percent ram.percent cpu load_1m load_5m load_15m node.role node.roles                                        cluster_manager name
127.0.0.1           16          95   0    0.04    0.24     0.15 dimr      cluster_manager,data,ingest,remote_cluster_client *               node-1
```

```
[root@centos7 vagrant]# curl -k -u admin:KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA https://localhost:9200/_cluster/health?pretty
{
  "cluster_name" : "wazuh-cluster",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "discovered_master" : true,
  "discovered_cluster_manager" : true,
  "active_primary_shards" : 7,
  "active_shards" : 7,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 100.0
}
```

```
[root@centos7 vagrant]# curl -k -u admin:KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA https://localhost:9200/_cat/shards
.opensearch-observability   0 p STARTED  0    208b 127.0.0.1 node-1
.opendistro_security        0 p STARTED 10    44kb 127.0.0.1 node-1
.kibana_1                   0 p STARTED  4  60.7kb 127.0.0.1 node-1
wazuh-alerts-4.x-2023.10.26 2 p STARTED 71 187.2kb 127.0.0.1 node-1
wazuh-alerts-4.x-2023.10.26 1 p STARTED 68 187.6kb 127.0.0.1 node-1
wazuh-alerts-4.x-2023.10.26 0 p STARTED 71 198.9kb 127.0.0.1 node-1
wazuh-monitoring-2023.43w   0 p STARTED  0    208b 127.0.0.1 node-1
```

</details>

<details><summary>Upgrade to 4.8.0</summary>

- Stop Filebeat and Wazuh dashboard services

```
[root@centos7 vagrant]# systemctl stop filebeat
[root@centos7 vagrant]# systemctl stop wazuh-dashboard
```

- Stop shard allocation

```
[root@centos7 vagrant]# curl -X PUT "https://localhost:9200/_cluster/settings"  -u admin:KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA -k -H 'Content-Type: application/json' -d'
> {
>   "persistent": {
>     "cluster.routing.allocation.enable": "primaries"
>   }
> }
> '
{"acknowledged":true,"persistent":{"cluster":{"routing":{"allocation":{"enable":"primaries"}}}},"transient":{}}
```

```
[root@centos7 vagrant]# curl -X POST "https://localhost:9200/_flush/synced" -u admin:KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA -k
{"_shards":{"total":7,"successful":7,"failed":0}}
```

- Stop Wazuh indexer service and upgrade

```
[root@centos7 vagrant]# systemctl stop wazuh-indexer
[root@centos7 vagrant]# yum install https://packages-dev.wazuh.com/warehouse/test/4.8/rpm/var/wazuh-indexer-4.8.0-wp2521.x86_64.rpm
Loaded plugins: fastestmirror
wazuh-indexer-4.8.0-wp2521.x86_64.rpm                                                                                                                                                                                                                                                                 | 743 MB  00:00:32     
Examining /var/tmp/yum-root-Dkr1w1/wazuh-indexer-4.8.0-wp2521.x86_64.rpm: wazuh-indexer-4.8.0-wp2520.x86_64
Marking /var/tmp/yum-root-Dkr1w1/wazuh-indexer-4.8.0-wp2521.x86_64.rpm as an update to wazuh-indexer-4.5.4-1.x86_64
Resolving Dependencies
--> Running transaction check
---> Package wazuh-indexer.x86_64 0:4.5.4-1 will be updated
---> Package wazuh-indexer.x86_64 0:4.8.0-wp2520 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

=============================================================================================================================================================================================================================================================================================================================
 Package                                                                   Arch                                                               Version                                                                   Repository                                                                                      Size
=============================================================================================================================================================================================================================================================================================================================
Updating:
 wazuh-indexer                                                             x86_64                                                             4.8.0-wp2520                                                              /wazuh-indexer-4.8.0-wp2521.x86_64                                                             1.0 G

Transaction Summary
=============================================================================================================================================================================================================================================================================================================================
Upgrade  1 Package

Total size: 1.0 G
Is this ok [y/d/N]: y
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : wazuh-indexer-4.8.0-wp2520.x86_64                                                                                                                                                                                                                                                                         1/2 
  Cleanup    : wazuh-indexer-4.5.4-1.x86_64                                                                                                                                                                                                                                                                              2/2 
  Verifying  : wazuh-indexer-4.8.0-wp2520.x86_64                                                                                                                                                                                                                                                                         1/2 
  Verifying  : wazuh-indexer-4.5.4-1.x86_64                                                                                                                                                                                                                                                                              2/2 

Updated:
  wazuh-indexer.x86_64 0:4.8.0-wp2520                                                                                                                                                                                                                                                                                        

Complete!
```

- Check Wazuh indexer service and cluster status

```
[root@centos7 vagrant]# systemctl status wazuh-indexer
● wazuh-indexer.service - Wazuh-indexer
   Loaded: loaded (/usr/lib/systemd/system/wazuh-indexer.service; enabled; vendor preset: disabled)
   Active: active (running) since Thu 2023-10-26 12:59:05 UTC; 4s ago
     Docs: https://documentation.wazuh.com
 Main PID: 8326 (java)
   CGroup: /system.slice/wazuh-indexer.service
           └─8326 /usr/share/wazuh-indexer/jdk/bin/java -Xshare:auto -Dopensearch.networkaddress.cache.ttl=60 -Dopensearch.networkaddress.cache.negative.ttl=10 -XX:+AlwaysPreTouch -Xss1m -Djava.awt.headless=true -Dfile.encoding=UTF-8 -Djna.nosys=true -XX:-OmitStackTraceInFastThrow -XX:+ShowCodeDetailsInExceptionM...

Oct 26 12:58:58 centos7 systemd[1]: Starting Wazuh-indexer...
Oct 26 12:58:59 centos7 systemd-entrypoint[8326]: WARNING: A terminally deprecated method in java.lang.System has been called
Oct 26 12:58:59 centos7 systemd-entrypoint[8326]: WARNING: System::setSecurityManager has been called by org.opensearch.bootstrap.OpenSearch (file:/usr/share/wazuh-indexer/lib/opensearch-2.10.0.jar)
Oct 26 12:58:59 centos7 systemd-entrypoint[8326]: WARNING: Please consider reporting this to the maintainers of org.opensearch.bootstrap.OpenSearch
Oct 26 12:58:59 centos7 systemd-entrypoint[8326]: WARNING: System::setSecurityManager will be removed in a future release
Oct 26 12:59:00 centos7 systemd-entrypoint[8326]: WARNING: A terminally deprecated method in java.lang.System has been called
Oct 26 12:59:00 centos7 systemd-entrypoint[8326]: WARNING: System::setSecurityManager has been called by org.opensearch.bootstrap.Security (file:/usr/share/wazuh-indexer/lib/opensearch-2.10.0.jar)
Oct 26 12:59:00 centos7 systemd-entrypoint[8326]: WARNING: Please consider reporting this to the maintainers of org.opensearch.bootstrap.Security
Oct 26 12:59:00 centos7 systemd-entrypoint[8326]: WARNING: System::setSecurityManager will be removed in a future release
Oct 26 12:59:05 centos7 systemd[1]: Started Wazuh-indexer.
```

```
[root@centos7 vagrant]# curl -k -u admin:KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA https://localhost:9200/_cat/nodes?v
ip        heap.percent ram.percent cpu load_1m load_5m load_15m node.role node.roles                                        cluster_manager name
127.0.0.1           22          96   3    0.44    0.27     0.18 dimr      cluster_manager,data,ingest,remote_cluster_client *               node-1
```

```
[root@centos7 vagrant]# curl -X PUT "https://localhost:9200/_cluster/settings" -u admin:KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA -k -H 'Content-Type: application/json' -d'
> {
>   "persistent": {
>     "cluster.routing.allocation.enable": "all"
>   }
> }
> '
{"acknowledged":true,"persistent":{"cluster":{"routing":{"allocation":{"enable":"all"}}}},"transient":{}}
```

```
[root@centos7 vagrant]# curl -k -u admin:KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA https://localhost:9200/_cat/nodes?v
ip        heap.percent ram.percent cpu load_1m load_5m load_15m node.role node.roles                                        cluster_manager name
127.0.0.1           22          96   1    0.37    0.26     0.17 dimr      cluster_manager,data,ingest,remote_cluster_client *               node-1
```

```
[root@centos7 vagrant]# curl -k -u admin:KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA https://localhost:9200/_cluster/health?pretty
{
  "cluster_name" : "wazuh-cluster",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "discovered_master" : true,
  "discovered_cluster_manager" : true,
  "active_primary_shards" : 9,
  "active_shards" : 9,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 100.0
}
```

```
[root@centos7 vagrant]# curl -k -u admin:KBwK9.Ten?MYHHGu8hs9VgTs*JNWZXSA https://localhost:9200/_cat/shards
.opensearch-observability        0 p STARTED  0    208b 127.0.0.1 node-1
.plugins-ml-config               0 p STARTED  1   3.8kb 127.0.0.1 node-1
.opensearch-sap-log-types-config 0 p STARTED            127.0.0.1 node-1
wazuh-monitoring-2023.43w        0 p STARTED  0    208b 127.0.0.1 node-1
wazuh-alerts-4.x-2023.10.26      0 p STARTED 71 198.9kb 127.0.0.1 node-1
wazuh-alerts-4.x-2023.10.26      1 p STARTED 68 187.6kb 127.0.0.1 node-1
wazuh-alerts-4.x-2023.10.26      2 p STARTED 71 187.2kb 127.0.0.1 node-1
.opendistro_security             0 p STARTED 10    44kb 127.0.0.1 node-1
.kibana_1                        0 p STARTED  4  32.3kb 127.0.0.1 node-1
```

- Upgrade Wazuh manager

```
[root@centos7 vagrant]# yum install https://packages-dev.wazuh.com/warehouse/test/4.8/rpm/var/wazuh-manager-4.8.0-wp2521.x86_64.rpm -y
Loaded plugins: fastestmirror
wazuh-manager-4.8.0-wp2521.x86_64.rpm                                                                                                                                                                                                                                                                 | 165 MB  00:00:07     
Examining /var/tmp/yum-root-Dkr1w1/wazuh-manager-4.8.0-wp2521.x86_64.rpm: wazuh-manager-4.8.0-wp2521.x86_64
Marking /var/tmp/yum-root-Dkr1w1/wazuh-manager-4.8.0-wp2521.x86_64.rpm as an update to wazuh-manager-4.5.4-1.x86_64
Resolving Dependencies
--> Running transaction check
---> Package wazuh-manager.x86_64 0:4.5.4-1 will be updated
---> Package wazuh-manager.x86_64 0:4.8.0-wp2521 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

=============================================================================================================================================================================================================================================================================================================================
 Package                                                                   Arch                                                               Version                                                                   Repository                                                                                      Size
=============================================================================================================================================================================================================================================================================================================================
Updating:
 wazuh-manager                                                             x86_64                                                             4.8.0-wp2521                                                              /wazuh-manager-4.8.0-wp2521.x86_64                                                             602 M

Transaction Summary
=============================================================================================================================================================================================================================================================================================================================
Upgrade  1 Package

Total size: 602 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : wazuh-manager-4.8.0-wp2521.x86_64                                                                                                                                                                                                                                                                         1/2 
warning: /var/ossec/etc/ossec.conf created as /var/ossec/etc/ossec.conf.rpmnew
  Cleanup    : wazuh-manager-4.5.4-1.x86_64                                                                                                                                                                                                                                                                              2/2 
  Verifying  : wazuh-manager-4.8.0-wp2521.x86_64                                                                                                                                                                                                                                                                         1/2 
  Verifying  : wazuh-manager-4.5.4-1.x86_64                                                                                                                                                                                                                                                                              2/2 

Updated:
  wazuh-manager.x86_64 0:4.8.0-wp2521                                                                                                                                                                                                                                                                                        

Complete!
```

- Check Wazuh manager status

```
[root@centos7 vagrant]# systemctl status wazuh-manager.service 
● wazuh-manager.service - Wazuh manager
   Loaded: loaded (/usr/lib/systemd/system/wazuh-manager.service; enabled; vendor preset: disabled)
   Active: active (running) since Thu 2023-10-26 13:02:07 UTC; 3min 17s ago
   CGroup: /system.slice/wazuh-manager.service
           ├─9120 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
           ├─9160 /var/ossec/bin/wazuh-authd
           ├─9175 /var/ossec/bin/wazuh-db
           ├─9189 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
           ├─9192 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
           ├─9195 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
           ├─9208 /var/ossec/bin/wazuh-execd
           ├─9223 /var/ossec/bin/wazuh-analysisd
           ├─9289 /var/ossec/bin/wazuh-syscheckd
           ├─9305 /var/ossec/bin/wazuh-remoted
           ├─9337 /var/ossec/bin/wazuh-logcollector
           ├─9388 /var/ossec/bin/wazuh-monitord
           └─9437 /var/ossec/bin/wazuh-modulesd

Oct 26 13:01:59 centos7 env[9061]: Started wazuh-db...
Oct 26 13:02:00 centos7 env[9061]: Started wazuh-execd...
Oct 26 13:02:01 centos7 env[9061]: Started wazuh-analysisd...
Oct 26 13:02:01 centos7 env[9061]: Started wazuh-syscheckd...
Oct 26 13:02:02 centos7 env[9061]: Started wazuh-remoted...
Oct 26 13:02:03 centos7 env[9061]: Started wazuh-logcollector...
Oct 26 13:02:04 centos7 env[9061]: Started wazuh-monitord...
Oct 26 13:02:05 centos7 env[9061]: Started wazuh-modulesd...
Oct 26 13:02:07 centos7 env[9061]: Completed.
Oct 26 13:02:07 centos7 systemd[1]: Started Wazuh manager.
```

- Upgrade Filebeat

```
[root@centos7 vagrant]# curl -s https://packages.wazuh.com/4.x/filebeat/wazuh-filebeat-0.2.tar.gz | sudo tar -xvz -C /usr/share/filebeat/module
wazuh/alerts/
wazuh/alerts/config/
wazuh/alerts/config/alerts.yml
wazuh/alerts/manifest.yml
wazuh/alerts/ingest/
wazuh/alerts/ingest/pipeline.json
wazuh/archives/
wazuh/archives/config/
wazuh/archives/config/archives.yml
wazuh/archives/manifest.yml
wazuh/archives/ingest/
wazuh/archives/ingest/pipeline.json
wazuh/module.yml
```

```
[root@centos7 vagrant]# curl -so /etc/filebeat/wazuh-template.json https://raw.githubusercontent.com/wazuh/wazuh/4.8.0/extensions/elasticsearch/7.x/wazuh-template.json
[root@centos7 vagrant]# chmod go+r /etc/filebeat/wazuh-template.json
```

- Check Filebeat status

```
[root@centos7 vagrant]# systemctl daemon-reload
[root@centos7 vagrant]# systemctl enable filebeat
[root@centos7 vagrant]# systemctl start filebeat
[root@centos7 vagrant]# filebeat test output
elasticsearch: https://127.0.0.1:9200...
  parse url... OK
  connection...
    parse host... OK
    dns lookup... OK
    addresses: 127.0.0.1
    dial up... OK
  TLS...
    security: server's certificate chain verification is enabled
    handshake... OK
    TLS version: TLSv1.2
    dial up... OK
  talk to server... OK
  version: 7.10.2
```

- Upgrade Wazuh dashboard

```
[root@centos7 vagrant]# yum install https://packages-dev.wazuh.com/warehouse/test/4.8/rpm/var/wazuh-dashboard-4.8.0-wp2521.x86_64.rpm -y
Loaded plugins: fastestmirror
wazuh-dashboard-4.8.0-wp2521.x86_64.rpm                                                                                                                                                                                                                                                               | 268 MB  00:00:12     
Examining /var/tmp/yum-root-Dkr1w1/wazuh-dashboard-4.8.0-wp2521.x86_64.rpm: wazuh-dashboard-4.8.0-wp2521.x86_64
Marking /var/tmp/yum-root-Dkr1w1/wazuh-dashboard-4.8.0-wp2521.x86_64.rpm as an update to wazuh-dashboard-4.5.4-1.x86_64
Resolving Dependencies
--> Running transaction check
---> Package wazuh-dashboard.x86_64 0:4.5.4-1 will be updated
---> Package wazuh-dashboard.x86_64 0:4.8.0-wp2521 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

=============================================================================================================================================================================================================================================================================================================================
 Package                                                                    Arch                                                              Version                                                                  Repository                                                                                       Size
=============================================================================================================================================================================================================================================================================================================================
Updating:
 wazuh-dashboard                                                            x86_64                                                            4.8.0-wp2521                                                             /wazuh-dashboard-4.8.0-wp2521.x86_64                                                            889 M

Transaction Summary
=============================================================================================================================================================================================================================================================================================================================
Upgrade  1 Package

Total size: 889 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : wazuh-dashboard-4.8.0-wp2521.x86_64                                                                                                                                                                                                                                                                       1/2 
warning: /etc/wazuh-dashboard/opensearch_dashboards.yml created as /etc/wazuh-dashboard/opensearch_dashboards.yml.rpmnew
  Cleanup    : wazuh-dashboard-4.5.4-1.x86_64                                                                                                                                                                                                                                                                            2/2 
  Verifying  : wazuh-dashboard-4.8.0-wp2521.x86_64                                                                                                                                                                                                                                                                       1/2 
  Verifying  : wazuh-dashboard-4.5.4-1.x86_64                                                                                                                                                                                                                                                                            2/2 

Updated:
  wazuh-dashboard.x86_64 0:4.8.0-wp2521                                                                                                                                                                                                                                                                                      

Complete!
```

- Check Wazuh dashboard status

```
[root@centos7 vagrant]# systemctl daemon-reload
[root@centos7 vagrant]# systemctl enable wazuh-dashboard
[root@centos7 vagrant]# systemctl start wazuh-dashboard
[root@centos7 vagrant]# systemctl status wazuh-dashboard.service 
● wazuh-dashboard.service - wazuh-dashboard
   Loaded: loaded (/etc/systemd/system/wazuh-dashboard.service; enabled; vendor preset: disabled)
   Active: active (running) since Thu 2023-10-26 13:05:16 UTC; 14s ago
 Main PID: 10751 (node)
   CGroup: /system.slice/wazuh-dashboard.service
           └─10751 /usr/share/wazuh-dashboard/node/fallback/bin/node --no-warnings --max-http-header-size=65536 --unhandled-rejections=warn /usr/share/wazuh-dashboard/src/cli/dist

Oct 26 13:05:19 centos7 opensearch-dashboards[10751]: {"type":"log","@timestamp":"2023-10-26T13:05:19Z","tags":["info","savedobjects-service"],"pid":10751,"message":"Waiting until all OpenSearch nodes are compatible with OpenSearch Dashboards before starting saved objects migrations..."}
Oct 26 13:05:19 centos7 opensearch-dashboards[10751]: {"type":"log","@timestamp":"2023-10-26T13:05:19Z","tags":["info","savedobjects-service"],"pid":10751,"message":"Starting saved objects migrations"}
Oct 26 13:05:19 centos7 opensearch-dashboards[10751]: {"type":"log","@timestamp":"2023-10-26T13:05:19Z","tags":["info","savedobjects-service"],"pid":10751,"message":"Detected mapping change in \"properties.visualization-visbuilder\""}
Oct 26 13:05:19 centos7 opensearch-dashboards[10751]: {"type":"log","@timestamp":"2023-10-26T13:05:19Z","tags":["info","savedobjects-service"],"pid":10751,"message":"Creating index .kibana_2."}
Oct 26 13:05:19 centos7 opensearch-dashboards[10751]: {"type":"log","@timestamp":"2023-10-26T13:05:19Z","tags":["info","savedobjects-service"],"pid":10751,"message":"Migrating .kibana_1 saved objects to .kibana_2"}
Oct 26 13:05:19 centos7 opensearch-dashboards[10751]: {"type":"log","@timestamp":"2023-10-26T13:05:19Z","tags":["info","savedobjects-service"],"pid":10751,"message":"Pointing alias .kibana to .kibana_2."}
Oct 26 13:05:19 centos7 opensearch-dashboards[10751]: {"type":"log","@timestamp":"2023-10-26T13:05:19Z","tags":["info","savedobjects-service"],"pid":10751,"message":"Finished in 165ms."}
Oct 26 13:05:19 centos7 opensearch-dashboards[10751]: {"type":"log","@timestamp":"2023-10-26T13:05:19Z","tags":["info","plugins-system"],"pid":10751,"message":"Starting [46] plugins: [usageCollection,opensearchDashboardsUsageCollection,opensearchDashboardsLegacy,mapsLegacy,share,opensearch...expressions,data,home,ap
Oct 26 13:05:19 centos7 opensearch-dashboards[10751]: {"type":"log","@timestamp":"2023-10-26T13:05:19Z","tags":["listening","info"],"pid":10751,"message":"Server running at https://0.0.0.0:443"}
Oct 26 13:05:19 centos7 opensearch-dashboards[10751]: {"type":"log","@timestamp":"2023-10-26T13:05:19Z","tags":["info","http","server","OpenSearchDashboards"],"pid":10751,"message":"http server running at https://0.0.0.0:443"}
Hint: Some lines were ellipsized, use -l to show in full.
```

</details>

- WUI is accesible, new WUI observed
- There are no errors seen in the logs that have not previously appeared in previous versions after the upgrade

---

### Notes

- The configuration of the Wazuh dashboard changes and is a forced change, due to the RPM management in files different from the original installed, if the original file has been modified, it is not updated in the upgrade process, this creates an error if the configuration is not manually updated (`/app/wazuh` -> `/app/wz-home`), suggestions:
   - [ ] Add a note in documentation so that it can be modified manually by users
   - [ ] Delegate the modification of the user's file to the upgrade process always
   - [ ] Modify the redirection, continue using `/app/wazuh`, but internally redirect with `/app/wz-wazuh`

**Comment by davidjiglesias** [comment]
LGTM!
